app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -627,9 +627,15 @@ class BrowserTabViewModel(
         }
     }
 
-    override fun iconReceived(icon: Bitmap) {
+    override fun iconReceived(url: String, icon: Bitmap) {
         val currentTab = tabRepository.liveSelectedTab.value ?: return
-        val url = currentTab.url ?: return
+        val currentUrl = currentTab.url ?: return
+        if (currentUrl != url) {
+            pixel.enqueueFire(AppPixelName.FAVICON_WRONG_URL_ERROR)
+            Timber.d("Favicon received for a url $url, different than the current one $currentUrl")
+            return
```

Comment:
why not store that favicon? I think it should be ok to store it if we don't use it for the current URL. That way we will increase our hit ratio for favicons. WDTY?

Reply:
We have created an internal task to persist possible best quality favicons if they belong to persisted site: https://app.asana.com/0/414730916066338/1200237930163733/f